couch_rs/src/database.rs

```diff
@@ -788,6 +788,43 @@ impl Database {
         }
     }
 
+    /// Bulk upsert a list of documents.
+    ///
+    /// This will first fetch the latest rev for each document that does not have a rev set. It
+    /// will then insert all documents into the database.
+    pub async fn bulk_upsert<T: TypedCouchDocument + Clone>(&self, mut docs: &mut Vec<T>)-> CouchResult<Vec<DocumentCreatedResult>>
+    {
+        // First collect all docs that do not have a rev set.
+        let mut docs_without_rev = vec![];
+        for (i, doc) in docs.iter().enumerate() {
+            if doc.get_rev().is_empty() && !doc.get_id().is_empty() {
+                docs_without_rev.push((doc.get_id().to_string(), i));
+            }
+        }
+
+        // Fetch the latest rev for the docs that do not have a rev set.
+        let ids_without_rev: Vec<String> = docs_without_rev.iter().map(|(id, _)| id.to_string()).collect();
```

Comment:
would it make sense to create/populate ids_without_rev inside the previous for loop, given we're already traversing all the documents once?

Reply:
...and maybe also check if ids_without_rev is empty before doing the get_bulk?

Reply:
sorry, forgot to mention, thanks for the PR!